src/components/TransactionsList.tsx

```diff
@@ -1,59 +1,81 @@
+          "x-citizen-id": props.location.state
+            ? props.location.state.citizenid
+            : getCitizenId()
+        }),
+      toError
+    )
+      .foldTaskEither(
+        apiError =>
+          fromLeft<Error, TypeofApiResponse<GetBPDTransactionsT>>(apiError),
+        apiResponse =>
+          fromEither(
+            apiResponse.mapLeft(err => {
+              return new Error(readableReport(err));
+            })
+          )
+      )
+      .mapLeft(_ => {
+        // TODO: Validation Error
+      })
+      .map(_ => {
+        if (_.status === 200) {
+          setResultdata(_.value);
+        }
+      })
+      .run()
+      .catch(_ => {
+        setResulterr(_.value);
+      });
+  }, []);
+
   return (
     <>
       <h3>{t("Transactions list")}</h3>
-
+      {resultErr && <div className="alert">Error: {resultErr}</div>}
       <div className="mt-3">
         <div className="row border-bottom border-dark py-2">
-          <div className="col-sm-2 font-weight-bold">Data ora</div>
-          <div className="col-sm-2 font-weight-bold">Importo</div>
-          <div className="col-sm-2 font-weight-bold">Cashback</div>
-          <div className="col-sm-3 font-weight-bold">HPAN</div>
-          <div className="col-sm-2 font-weight-bold">Circuit</div>
+          <div className="col-sm-2 font-weight-bold">{t("Datetime")}</div>
+          <div className="col-sm-3 font-weight-bold">{t("Acquirer")}</div>
+          <div className="col-sm-2 font-weight-bold">{t("Circuit name")}</div>
+          <div className="col-sm-2 font-weight-bold">{t("Amount")}</div>
+          <div className="col-sm-2 font-weight-bold">{t("HPAN")}</div>
           <div className="col-sm-1 font-weight-bold"></div>
         </div>
-        {[...Array(30)].map(index => (
-          <div
-            key={index}
-            className={`TransactionsList__row row py-2 ${
-              index % 2 === 1 ? "bg-light" : ""
-            }`}
-          >
-            <div className="col-sm-2">01/12/20 12:23</div>
-            <div className="col-sm-2 ">300,21</div>
-            <div className="col-sm-2 ">30,00</div>
-            <div className="col-sm-3 ">*432</div>
-            <div className="col-sm-2 ">Visa </div>
-            <div className="col-sm-1 p-0">
-              <img
-                src={viewIcon}
-                className="mr-3 cursor-pointer"
-                height="24"
-                onClick={() => {
-                  openDetail(index);
-                }}
-              />
-              <span className="badge badge-primary">RAW</span>
-            </div>
-            <div
-              className="col-12 d-none TransactionsList__detail"
-              id={`detail${index}`}
-            >
-              {t("Transaction details")}
-            </div>
-          </div>
-        ))}
+        {resultData &&
+          resultData.transactions.map((el: BPDTransaction, index: number) => (
+            <Transaction el={el} index={index} key={index} />
```

Comment:
Generally is safe to use index as key only if the list is static (no update) and there is no filtering/re-ordering. It this the case?

Reply:
thanks for the tip, but -for now- the list is not filtered or reordered